Source/EventFlow/EventStores/Files/FilesEventPersistence.cs

```diff
@@ -167,20 +168,29 @@ public class EventStoreLog
                     committedDomainEvents.Add(fileEventData);
                 }
 
-                using (var streamWriter = File.CreateText(_logFilePath))
+                //only allow a single write access to the _logFilePath file at one time
+                await _logFileSemaphore.WaitAsync().ConfigureAwait(false);
```

Comment:
There's an `AsyncLock` in the EventFlow code base that supports cancellation and the `using` pattern, please use that.